### PR TITLE
Misc. enhancement

### PR DESCRIPTION
**Pull request by nonprofittechy**
* Change some direct key references to .get, #47
* #40 Skip filing_component question when it's define
* #39 Use the case type / category from the found case
* Make the case search a little more robust
* Use the case search in the any filing interview #30

### CODE REVIEW

File: docassemble/EFSPIntegration/data/questions/case_search.yml [hunk 1]
```diff
@@ -16,7 +16,10 @@ code: |
     somebody.name.first
     court_id    
     get_cases_response
-    case_tracking_id = case_tracking_choice
+    if case_tracking_choice is None:
+      case_tracking_id
```
Comment (BryceStevenWilley) [inline docassemble/EFSPIntegration/data/questions/case_search.yml:20]: I think this sets the tracking id (which is what I've been calling the Tyler-internal GUIDs for cases / filings) so we can continue on in the Any filing interview, right? If so, I think it means that the user can't find the case, and is likely non-indexed meaning there isn't actually a tracking id. They should have a docket number, but that'll be a different question / variable.

File: docassemble/EFSPIntegration/data/questions/any_filing_interview.yml [hunk 2]
```diff
@@ -201,6 +218,18 @@ fields:
     code: |
       filing_component_options
 ---
+if: |
+  exactly_one_required_filing_component(filing_component_options)
+code: |
+  lead_doc.filing_component_code = next(iter(filing_component_options))[0]
```
Comment (BryceStevenWilley) [inline docassemble/EFSPIntegration/data/questions/any_filing_interview.yml:224]: Nice, I like this implementation a lot.

**Review by BryceStevenWilley — APPROVED** [review]
LGTM. I'm going to merge, but I would like some clarification about how you updated the case search stuff. Maybe I'll understand it after playing around with it though.

File: docassemble/EFSPIntegration/data/questions/case_search.yml (hunk 1 above)
Comment (nonprofittechy) [reply docassemble/EFSPIntegration/data/questions/case_search.yml:20]: agreed, this is just slightly better than setting the case_tracking_id to None. We still need a full fix for it

**Pull request merged**